components/react/hooks/src/useDynamicHeightCollapsible/index.js

```diff
@@ -0,0 +1,17 @@
+import {useState, useCallback} from 'react'
+import useToggle from '../useToggle'
+
+export default function useDynamicHeightCollapsible(getContentHeighFn) {
+  const [isCollapsed, toggleCollapsible] = useToggle(true)
+  const [contentHeight, setContentHeight] = useState(null)
+  const heightRecalculator = useCallback(() => {
+    const newContentHeight = getContentHeighFn()
+    setContentHeight(newContentHeight)
+  }, [getContentHeighFn])
+
+  return [
```

Comment:
This hook will return more than a state and more than a handler. In this case, it's better to return an object of named properties, since the order of the array make the restructuring error prone. I'd suggest something like:
```js
return {
  isCollapsed,
  height,
  toggle,
  heightRecalculator
}
```

Reply:
done!